Index: iridium-browser/ui/views/widget/desktop_aura/desktop_screen_x11.cc
===================================================================
--- iridium-browser.orig/ui/views/widget/desktop_aura/desktop_screen_x11.cc	2015-02-02 15:03:18.000000000 +0100
+++ iridium-browser/ui/views/widget/desktop_aura/desktop_screen_x11.cc	2015-02-02 15:03:23.000000000 +0100
@@ -8,8 +8,10 @@
 #include <X11/Xlib.h>
 
 #ifdef USE_GLIB
+extern "C" {
 #include <gio/gio.h>
 #include <glib.h>
+}
 #endif
 
 // It clashes with our RootWindow.
