Index: iridium-ubuntu-build/chrome/app/chromium_strings.grd
===================================================================
--- iridium-ubuntu-build.orig/chrome/app/chromium_strings.grd
+++ iridium-ubuntu-build/chrome/app/chromium_strings.grd
@@ -195,7 +195,7 @@ If you update this file, be sure also to
       <if expr="not chromeos">
         <!-- Browser Window Title Format -->
         <message name="IDS_BROWSER_WINDOW_TITLE_FORMAT" desc="The format for titles displayed in tabs and popup windows">
-          <ph name="PAGE_TITLE">$1<ex>Google</ex></ph> - Chromium
+          <ph name="PAGE_TITLE">$1<ex>Google</ex></ph> - Iridium
         </message>
       </if>
       <if expr="chromeos">
