Index: iridium-ubuntu-build/chrome/BUILD.gn
===================================================================
--- iridium-ubuntu-build.orig/chrome/BUILD.gn
+++ iridium-ubuntu-build/chrome/BUILD.gn
@@ -1793,8 +1793,8 @@ if (is_linux) {
       filename = "google-chrome"
       confdir = "google-chrome"
     } else {
-      name = "Chromium"
-      filename = "chromium-browser"
+      name = "Iridium"
+      filename = "iridium-browser"
       confdir = "chromium"
     }
 
