Index: iridium-ubuntu-build/base/metrics/histogram.h
===================================================================
--- iridium-ubuntu-build.orig/base/metrics/histogram.h
+++ iridium-ubuntu-build/base/metrics/histogram.h
@@ -81,7 +81,7 @@
 #include "base/macros.h"
 #include "base/metrics/bucket_ranges.h"
 #include "base/metrics/histogram_base.h"
-#if defined(OS_CHROMEOS) || defined(OS_IOS) || defined(OS_WIN)
+#if defined(OS_CHROMEOS) || defined(OS_IOS) || defined(OS_WIN) || defined(OS_LINUX)
 // TODO(asvitkine): Migrate callers to to include this directly and remove this.
 // Note: Incrementally migrating platforms as they become clean.
 #include "base/metrics/histogram_macros.h"
