Index: iridium-ubuntu-build/chrome/app/resources/locale_settings.grd
===================================================================
--- iridium-ubuntu-build.orig/chrome/app/resources/locale_settings.grd
+++ iridium-ubuntu-build/chrome/app/resources/locale_settings.grd
@@ -207,7 +207,7 @@
 
       <!-- The URL for the the Welcome to Chrome page shown on first run. -->
       <message name="IDS_WELCOME_PAGE_URL" translateable="false">
-        https://www.google.com/intl/[GRITLANGCODE]/chrome/browser/welcome.html
+        https://iridiumbrowser.de/welcome/[GRITLANGCODE]/welcome.html
       </message>
 
       <!-- The language dependent download view size. -->
Index: iridium-ubuntu-build/chrome/browser/first_run/first_run.cc
===================================================================
--- iridium-ubuntu-build.orig/chrome/browser/first_run/first_run.cc
+++ iridium-ubuntu-build/chrome/browser/first_run/first_run.cc
@@ -584,6 +584,7 @@ void SetShouldShowWelcomePage() {
 }
 
 bool ShouldShowWelcomePage() {
+  return false;
   bool retval = g_should_show_welcome_page;
   g_should_show_welcome_page = false;
   return retval;
