Index: iridium-ubuntu-build/chrome_elf/chrome_elf_constants.cc
===================================================================
--- iridium-ubuntu-build.orig/chrome_elf/chrome_elf_constants.cc
+++ iridium-ubuntu-build/chrome_elf/chrome_elf_constants.cc
@@ -10,7 +10,7 @@
 // depend on BrowserDistribution. http://crbug.com/577820
 #define PRODUCT_STRING_PATH L"Google\\Chrome"
 #elif defined(CHROMIUM_BUILD)
-#define PRODUCT_STRING_PATH L"Chromium"
+#define PRODUCT_STRING_PATH L"Iridium"
 #else
 #error Unknown branding
 #endif
