Index: iridium-ubuntu-build/chrome/BUILD.gn
===================================================================
--- iridium-ubuntu-build.orig/chrome/BUILD.gn
+++ iridium-ubuntu-build/chrome/BUILD.gn
@@ -253,7 +253,7 @@ if (!is_android && !is_mac) {
         "//chrome/common:features",
       ]
 
-      ldflags = [ "-pie", "-Wl,--no-keep-memory" ]
+      ldflags += [ "-pie" ]
 
       if (use_pango || use_cairo) {
         # Needed for chrome_main.cc initialization of libraries.
