Index: iridium-browser/third_party/openmax_dl/dl/sp/api/omxSP.h
===================================================================
--- iridium-browser.orig/third_party/openmax_dl/dl/sp/api/omxSP.h	2015-02-06 10:38:57.000000000 +0100
+++ iridium-browser/third_party/openmax_dl/dl/sp/api/omxSP.h	2015-02-06 10:40:12.000000000 +0100
@@ -2554,8 +2554,8 @@
 #define omxSP_FFTInv_CCSToR_F32 omxSP_FFTInv_CCSToR_F32_Sfs
 #else
 /* Build-time non-NEON selection. */
-#define omxSP_FFTFwd_RToCCS_F32 omxSP_FFTFwd_CCSToR_F32_vfp
-#define omxSP_FFTInv_CCSToR_F32 omxSP_FFTInv_CCSToR_F32_vfp
+#define omxSP_FFTFwd_RToCCS_F32 omxSP_FFTFwd_RToCCS_F32_Sfs_vfp
+#define omxSP_FFTInv_CCSToR_F32 omxSP_FFTInv_CCSToR_F32_Sfs_vfp
 #endif  /* defined(DL_ARM_NEON_OPTIONAL) */
 
 #if defined(DL_ARM_NEON_OPTIONAL) || !defined(DL_ARM_NEON)
